### PR TITLE
machine: Pass machine, user, command values to polkit on OpenMachineShell

### PR DESCRIPTION
**Pull request by intgr**
This allows more granular access control in PolicyKit rules, similar to /etc/sudoers, for polkit actions:
* org.freedesktop.machine1.host-shell
* org.freedesktop.machine1.shell

Example configuration (e.g. `/etc/polkit-1/rules.d/50-ego-machinectl.rules`)
```js
polkit.addRule(function(action, subject) {
    if (action.id == "org.freedesktop.machine1.host-shell") {
        polkit.log("host-shell: " + action);
    }
    if (action.id == "org.freedesktop.machine1.host-shell"
        && action.lookup("user") == "ego"
        && subject.user == "marti") {
            return polkit.Result.YES;
    }
});
```

Log message from the `polkit.log()` call:

> polkitd[757]: /etc/polkit-1/rules.d/50-ego-machinectl.rules:3: host-shell: [Action id='org.freedesktop.machine1.host-shell' **machine='.host' user='ego' command='/bin/sh'**]

### CODE REVIEW

**Comment by intgr** [comment]
Hmm, actually it makes sense to rename `command` to `program`, like pkexec does with the polkit action `org.freedesktop.policykit.exec`. (https://www.freedesktop.org/software/polkit/docs/0.106/pkexec.1.html#pkexec-variables)

pkexec passes other variables too (command_line, user.gecos, user.display). I have no need for those, but I can replicate them if reviewers deem that useful.

File: src/machine/machine-dbus.c [hunk 1]
```diff
@@ -628,11 +628,17 @@ int bus_machine_method_open_shell(sd_bus_message *message, void *userdata, sd_bu
         if (!strv_env_is_valid(env))
                 return sd_bus_error_setf(error, SD_BUS_ERROR_INVALID_ARGS, "Invalid environment assignments");
 
+        const char *details[7] = {
```
Comment (keszybz) [inline src/machine/machine-dbus.c]: I'd prefer to leave the array size out, and have NULL as an explicit terminator below. That "7" is a bit surprising and having a terminator makes it less likely to be messed up on updates.

Comment (intgr) [reply src/machine/machine-dbus.c]: Thanks, good idea.

**Comment by intgr** [comment]
Rebased & updated per keszybz's comment & also renamed `command`->`program` per my previous comment.

**Comment by keszybz** [comment]
@poettering do we want this in this release or next?

**Comment by poettering** [comment]
lgtm, but I think we should do this after the release, i really have no idea how this affects existing policykit policies, or even selinux and such. Let's better be safe than sorry.

**Comment by intgr** [comment]
Is the "bionic-i386" check failure OK or should I look into it?

**Comment by keszybz** [comment]
It timed out, that's almost certainly unrelated.

**Pull request merged**